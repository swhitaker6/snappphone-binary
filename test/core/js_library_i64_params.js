TestLibrary = {
  jscall__deps: i53ConversionDeps,
  jscall__sig: 'ij',
  jscall: function({{{ defineI64Param('foo') }}}) {
    {{{ receiveI64ParamAsI53('foo', `(err('overflow'), ${makeReturn64('42')})`) }}}
    err('js:got:       ' + foo);
    if (foo < 0)
      var rtn = Math.ceil(foo / 2);
    else
      rtn = Math.floor(foo / 2);
    err('js:returning: ' + rtn);
    return {{{ makeReturn64('rtn') }}};
  },
}

mergeInto(LibraryManager.library, TestLibrary);
