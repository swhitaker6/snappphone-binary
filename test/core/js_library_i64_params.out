checking: 42
js:got:       42
js:returning: 21
got:      21
expected: 21
checking: 4503599627370496
js:got:       4503599627370496
js:returning: 2251799813685248
got:      2251799813685248
expected: 2251799813685248
checking: -4503599627370496
js:got:       -4503599627370496
js:returning: -2251799813685248
got:      -2251799813685248
expected: -2251799813685248
checking: 9007199254740992
js:got:       9007199254740992
js:returning: 4503599627370496
got:      4503599627370496
expected: 4503599627370496
checking: -9007199254740992
js:got:       -9007199254740992
js:returning: -4503599627370496
got:      -4503599627370496
expected: -4503599627370496
checking: 9007199254740993
overflow
got:      42
checking: -9007199254740993
overflow
got:      42
done
